cmd/vela-kubernetes/apply.go

```diff
@@ -59,7 +59,10 @@ func (a *Apply) Command(c *Config, file string) *exec.Cmd {
 	flags = append(flags, "apply")
 
 	// add flag for dry run from provided apply dry run
-	flags = append(flags, fmt.Sprintf("--dry-run=%t", a.DryRun))
+	if a.DryRun {
+		flags = append(flags, fmt.Sprintf("--dry-run=server"))
+
+	}
```

Comment:
what's the significance of this change?

Reply:
This change  updates deprecated flag `--dry-run`  `kubectl` and changes it to `--dry-run=server`, which performs server side validation. 
Using server side dry-run is preferable, because it addresses these [challenges](https://kubernetes.io/blog/2019/01/14/apiserver-dry-run-and-kubectl-diff/#challenges):
>- While compilers and linters do a good job to detect errors in pull-requests for code, a good validation is missing for Kubernetes configuration files. The existing solution is to run kubectl apply --dry-run, but this runs a local dry-run that doesn't talk to the server: it doesn't have server validation and doesn't go through validating admission controllers. As an example, Custom resource names are only validated on the server so a local dry-run won't help.
>- It can be difficult to know how your object is going to be applied by the server for multiple reasons:
>    * Defaulting will set some fields to potentially unexpected values,
>    * Mutating webhooks might set fields or clobber/change some values.
>    * Patch and merges can have surprising effects and result in unexpected objects. For example, it can be hard to know how lists are going to be ordered once merged.

Reply:
awesome, i appreciate the context 
could we remove this extra newline?
```suggestion
	if a.DryRun {
		flags = append(flags, fmt.Sprintf("--dry-run=server"))
	}
```